src/bot/snoopiBot.jl

```diff
@@ -0,0 +1,109 @@
+
+################################################################
+"""
+    @snoopiBot config::BotConfig snoopScript
+
+macro that generates precompile files and includes them in the package. Calls other bot functions.
+
+# Examples
+
+`@snoopiBot` the examples that call the package functions.
+
+```julia
+using SnoopCompile
+
+@snoopiBot "MatLang" begin
+  using MatLang
+  examplePath = joinpath(dirname(dirname(pathof(MatLang))), "examples")
+  include(joinpath(examplePath,"Language_Fundamentals", "usage_Entering_Commands.jl"))
+  include(joinpath(examplePath,"Language_Fundamentals", "usage_Matrices_and_Arrays.jl"))
+  include(joinpath(examplePath,"Language_Fundamentals", "Data_Types", "usage_Numeric_Types.jl"))
+end
+```
+"""
+macro snoopiBot(config::BotConfig, snoopScript)
```

Comment:
Macro names like `@snoopiBot`, `@snoopiBench` should ideally also respect the style guide line. For someone coming from Julia, the capitalization will look quite weird (almost like it is constructing a type, but not really). Lower Camel Case is never used in Julia

Reply:
Yeah, I probably should have insisted on changing these, I just felt guilty about letting this sit for so long. @aminya, this has come up so often that I think you really just need to sit down and carefully read the entire style guide and start embracing it in your code.

Reply:
Sorry about that, I have started to use `_` instead of capitalization for functions.

As a solution, we can create aliases if you want to but keep the old function names just for compatibility.

```
export snoopi_bot, snoopi_bench, snoopiBot, snoopiBench 
const snoopiBot = snoopi_bot
const snoopiBench = snoopi_bench
```

Reply:
You can probably use `Base.@deprecate_binding` to issue a warning about the old names.

Reply:
A patch was created 
https://github.com/timholy/SnoopCompile.jl/pull/66